tools/generate_tables_sql.php

```diff
@@ -68,26 +68,31 @@
 
             //generate specific column definitions for specific types of HTML elements
             default:
-                if($bits[1] == "") {
+                if((array_key_exists(1,$bits) ? $bits[1] : "") == "") {
                     continue;
                 }
                 if($bits[0]=="select"){
```

Comment:
I would move the $bits[0] checks to the top, above the switch, so they don't have to be done in each if statement.